pkg/agent/run.go

```diff
@@ -123,41 +121,31 @@ func Run(cmd *cobra.Command, args []string) {
 			logs.Log.Printf("failed to start %q data gatherer %q: %v", kind, dgConfig.Name, err)
 		}
 
-		// bootCtx is a context with a timeout to allow the informer 5
-		// seconds to perform an initial sync. It may fail, and that's fine
-		// too, it will backoff and retry of its own accord. Initial boot
-		// will only be delayed by a max of 5 seconds.
-		bootCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
-		defer cancel()
-
-		// wait for the informer to complete an initial sync, we do this to
-		// attempt to have an initial set of data for the first upload of
-		// the run.
-		if err := newDg.WaitForCacheSync(bootCtx.Done()); err != nil {
```

Comment:
Instead of waiting for each dataGatherer before starting to the next, we should first start all dataGatherers and loop again to wait for them to finish.